### ISSUE TITLE
Database Maintenance job should use FILLFACTOR = 80

### ISSUE THREAD

**Issue by cabal95** [SparkDevNetwork/Rock#2271]
### Prerequisites
* [X] Put an X between the brackets on this line if you have done all of the following:
    * Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?
    * Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?
    * Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ASparkDevNetwork%2FSlack) to see if your bug or enhancement is already reported?

### Description

The new Database Maintenance job uses a stored procedure to reorganize and rebuild indexes. However it uses the default FILLFACTOR value, which is 0. A value of 0 == 100 according to the MSDN docs, meaning when the index is rebuilt it is filled to 100% from the get go.

This means that indexes will begin to fragment almost immediately. Using 80% seems to be a recommended value and a quick test on our production server shows dramatic improvement on the PageView table indexes, which historically have had *horrible* fragmentation.

### Steps to Reproduce

1. Rebuild indexes on a table like PageView and watch the fragmentation % grow almost immediately.
2. Rebuild indexes again with FILLFACTOR = 80 and watch the frag % not grow.

**Expected behavior:** 

My database should be fast.

**Actual behavior:** 

My database isn't as fast as it could be.

### Versions

* **Rock Version:** git (pre-release code)
* **Client Culture Setting:** en-us

**Comment by edmistj** [comment]
@MikeDPeterson Can you add a migration to the roll-up for this.

**Comment by dcs619** [comment]
If a migration is needed to edit the maintenance job, can you also include this command option in the [script](https://github.com/SparkDevNetwork/Rock/blob/develop/Rock.Migrations/Migrations/201701022126386_DataMaintenanceJob.sql#L66)?

Online index rebuilds are supported in SQL Enterprise and Azure SQL, so end users are never affected during maintenance.

```
DECLARE @CommandOption varchar(100) = 'FILLFACTOR = 80';

-- SQL Enterprise is 3 and Azure is 5
IF ( SERVERPROPERTY('EngineEdition') = 3 OR SERVERPROPERTY('EngineEdition') = 5 )
BEGIN
	SELECT @CommandOption = @CommandOption + ', ONLINE = ON';
END

...

IF (@FragmentationPercent > @MinFragmentationRebuild)
    SET @SqlCommand = N'ALTER INDEX [' + @IndexName + N'] ON [' +  @SchemaName + N'].[' + @TableName + '] REBUILD WITH (' + @CommandOption + ')';
ELSE
    SET @SqlCommand = N'ALTER INDEX [' + @IndexName + N'] ON [' +  @SchemaName + N'].[' + @TableName + '] REORGANIZE';


```

**Comment by dcs619** [comment]
Fixed in v7, 69e0840852b5f6c7672f2742af3022acd23997d9
